packages/js/product-editor/src/components/attribute-term-input-field/create-attribute-term-modal.tsx

```diff
@@ -92,21 +92,19 @@ export const CreateAttributeTermModal: React.FC<
 	return (
 		<Modal
 			title={ __( 'Create attribute', 'woocommerce' ) }
-			onRequestClose={ (
-				event:
-					| React.KeyboardEvent< Element >
-					| React.MouseEvent< Element >
-					| React.FocusEvent< Element >
-			) => {
-				event.stopPropagation();
+			onRequestClose={ ( event ) => {
+				event?.stopPropagation();
 				onCancel();
 			} }
 			className="woocommerce-create-attribute-term-modal"
 		>
-			<Form< Partial< ProductAttributeTerm > >
+			<Form<
+				Pick< ProductAttributeTerm, 'name' | 'slug' | 'description' >
```

Comment:
@louwie17 what do you think about this change? Does it make sense?

Reply:
Yeah, this works for me as well, given we are only handling name, slug, and description within the form anyhow.